lib/mix/lib/mix/tasks/escript.install.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Mix.Tasks.Escript.Install do
+  use Mix.Task
+
+  @shortdoc "Install an escript locally"
+
+  @moduledoc """
+  Install an escript locally.
+
+  If no argument is supplied but there is an escript in the project's root directory
+  (created with `mix escript.build`), then the escript will be installed
+  locally. For example:
+
+      mix do escript.build, escript.install
+
+  The argument is usually a package name and optionally a version:
+
+      mix escript.install phoenix
+      mix escript.install phoenix 0.1.2
+
+  It could also be a remote repository:
+
+      mix escript.install git://example.com/foo.git          # infers Git from the URL
+      mix escript.install https://example.com/foo.git        # infers Git from the URL
+      mix escript.install github.com/example/foo             # fetches git://github.com/example/example.git
+      mix escript.install https://github.com/example/foo     # fetches https://github.com/example/example.git
+      mix escript.install --scm=git example.com/foo          # fetches git://example.com/example
+      mix escript.install --scm=git https://example.com/foo  # fetches https://example.com/example
```

Comment:
In the case of `https://example.com/foo.git` above, Git would be inferred from the trailing `.git` part.

After some consideration I see the appeal of sticking with your proposal because of its similarity with dep syntax. It doesn't define a few things though. What is your take on these cases?

```
escript.install ./local_file        # ./ is used to distinguish from a hex package
escript.install path/to/local_file  # cannot be a hex package, so it can imply a local path
escript.install path path/to/local_repo  # ugly

# What about this? Should we forbid fetching remote scripts?
escript.install ??? https://example.com/escript_file
```

I realize that in the common case hex packages should be preferred, but we still have to define what happens with the other cases.

The ambiguity of supporting a prebuilt escript could be solved with a dedicated task:

```
escript.install-file local_file
escript.install-file path/to/local_file
escript.install-file http://example.com/escript

# or a flag

escript.install --file local_file
# etc...
```

It may not be all too important to be able to install prebuilt escripts, but leaving it out makes it feel incomplete.

What do you think, @josevalim, @ericmj ?